### ISSUE TITLE
Post Flair Feature

### ISSUE THREAD

**Issue by MrLuxuri** [LemmyNet/lemmy#1456]
I recently moved from Reddit to Lemmy fully and one of the things I found missing was Post Flair, it is a very helpful feature that can be used to filter and view posts, it doesn't necessarily need to work exactly like Reddit but somewhat similar.

**Comment by dessalines** [comment]
You mean community specific user flair? Because we already have different user display names.

**Comment by MrLuxuri** [comment]
I do know we can change our name but there should be a flair where you can name it CEO for example and add it to yourself, while making other user flair like staff or something. It would help categorize mods in a community, you can also use user fairs to name yourself beta tester, dev and etc. Post flair is when you name them question, discussion and etc. These flairs are like to categorize _like hashtag_ 

If you use reddit, you would know what I mean:)

**Comment by heylix** [comment]
> Post Flair, it is a very helpful feature that can be used to filter and view posts

I agree; currently there are a few questions in `!linux` which may benefit from a "resolved" flair. This could also make filtering easier, as in Reddit for example you can exclude flairs with GET parameters in the URL so you only get shown the flairs you want.

I don't really like the name "flair" tbh, it's more like a tag when you think about it. The communities themselves could have tags, too. There may be exclusive tags so you can have only the flair "open" or "resolved" but not both on a post tagged "question", but you can't have the "resolved" tag on a post tagged "news" (that's very complicated and I'd settle for a single tag first).

**Comment by dessalines** [comment]
We have a different issue for post tags: #317

**Comment by not-layla** [comment]
Just want to make sure we're on the same page on this one, before we start giving this a proper look.

User flairs on Reddit are scoped by subreddit.

If we were to work on this issue, we'd want them to be a bit more flexible than that, i.e. also have the concept of a user flair that can be site-wide. Tim over in #2383 talked about "[pinned flairs](https://github.com/LemmyNet/lemmy/issues/2383#issuecomment-1206875376)", would this be something you guys are interested in?

**Comment by dessalines** [comment]
We don't have to follow reddit with flairs being attached to the community... but I could see both instance and community flairs being useful.

I'm not sure if there needs to be a pinned at all, if they're all going to show next to your name anyway. Otherwise what's the point, you could just put that in your bio.

**Comment by Nutomic** [comment]
I dont have any strong opinion on this. But it seems like a relatively complex feature, so it would be a good idea to write a proposal document first, before starting to implement it. Basically describing how it should work, how flairs get configured and chosen by users, also how they are federated.

**Comment by kazarnowicz** [comment]
> User flairs on Reddit are scoped by subreddit.
> 
> If we were to work on this issue, we'd want them to be a bit more flexible than that, i.e. also have the concept of a user flair that can be site-wide. Tim over in #2383 talked about "[pinned flairs](https://github.com/LemmyNet/lemmy/issues/2383#issuecomment-1206875376)", would this be something you guys are interested in?

I'd like to chip in here and say that user flairs per community is valuable, and can help with moderation, in my experience. We took advantage of this on Reddit for our community (men over 30 in the LGBT community): since we have members ranging from 20-80, we use flairs indicating the age range of the poster/commenter - this helps the community to know if they're talking to someone who (for example) lived through the AIDS epidemic in the 80s/90s.

**Comment by porl** [comment]
> I'd like to chip in here and say that user flairs per community is valuable, and can help with moderation, in my experience. We took advantage of this on Reddit for our community (men over 30 in the LGBT community): since we have members ranging from 20-80, we use flairs indicating the age range of the poster/commenter - this helps the community to know if they're talking to someone who (for example) lived through the AIDS epidemic in the 80s/90s.

Similar story here. I'm (for now I guess) a mod of r/judo and we use "rank" flairs to allow people to have an idea of the relative experience of those giving advice.

Adding to the feature, it would be nice to have both user-selectable (or editable) options, as well as "mod approved only" ones that could be set for people by the mods of a community. In another reddit sub I'm involved with, this is used often for people to show they actually *are* who they say they are (company representatives etc) after they have gone through a verification process with the mods.

**Comment by trymeouteh** [comment]
Would like to see a flair/tagging feature for posts too.

**Comment by uroybd** [comment]
How will it interop with ActivityPub. I guess, if we represent flairs as tags it will work well.

**Comment by OdinsHat** [comment]
Just to throw some examples of subreddit-specific user flairs on Reddit and how they provide a useful feature:

**Anime user profiles:** https://old.reddit.com/r/anime/comments/17pysqw/what_is_the_best_ending_to_a_longrunning_anime/

**Country of User:**
* https://old.reddit.com/r/2westerneurope4u/comments/17py61n/explain_this_brits/
* https://old.reddit.com/r/europe/comments/17ie56a/meanwhile_in_czechia/

**US State of User:**
* https://old.reddit.com/r/AskAnAmerican/comments/17ocq7u/do_americans_find_it_inappropriate_to_brush_your/

All of these encourage engagement by the userbase.

**Comment by SocietasEvanescentes** [comment]
As a member of a browser community, it would be useful for employees, CEO, beta testers, etc. to identify themselves to help users.

**Comment by dullbananas** [comment]
These would be useful, especially for competing against Instagram:

* Allow user to have multiple flairs in a community
* On profile page, show all flairs that you both have
* Ability to find users that have one or more specified flairs
